src/utils/connectMongo.ts

```diff
@@ -33,3 +29,64 @@ export async function connectMongo(server: Server) {
     server.setHealthStatus(HealthCheckResponse.ServingStatus.NOT_SERVING);
   });
 }
+
+/**
+ * Attempts to connect to mongo connectionRetries times.
+ * Waits reconnectTimeout ms bewteen attempts.
+ * @param server - the server trying to connect.
+ */
+async function startConnectionAttempts(server: Server) {
+  const retries = parseInt(config.mongo.connectionRetries, 10);
+  const timeout = parseInt(config.mongo.reconnectTimeout, 10);
+
+  for (let i = 1; i <= retries; i++) {
+    const connectionRes: { success: boolean; error: Error | null } = await connect();
+
+    // if mongo connection attempt has failed
+    if (!connectionRes.success) {
+      log(Severity.ERROR, `connection retry (${i}/${retries}) ${connectionRes.error}`, 'connectDB', getCurrTraceId(), {
+        errMsg: connectionRes.error?.message,
+        stack: connectionRes.error?.stack,
+      });
+
+      server.setHealthStatus(HealthCheckResponse.ServingStatus.NOT_SERVING);
+      await sleep(timeout);
+    } else {
+      log(Severity.INFO, `connected to ${config.mongo.connectionString}`, 'connectDB');
+      server.setHealthStatus(HealthCheckResponse.ServingStatus.SERVING);
+      break;
+    }
+  }
+}
+
+/**
+ * Connects to mongo with mongoConnectionString.
+ */
+async function connect(): Promise<{ success: boolean; error: Error | null }> {
+  let success: boolean = false;
+  let error = null;
+
+  await mongoose.connect(
+    config.mongo.connectionString,
+    {
+      useCreateIndex: true,
+      useNewUrlParser: true,
+      useFindAndModify: false,
+      useUnifiedTopology: true,
+    },
+    async (err) => {
+      if (!err) {
+        success = true;
+      } else {
+        success = false;
+        error = err;
+      }
+    }
+  );
+
+  return { success, error };
```

Comment:
Maybe this will be prettier: (don't commit through here)
```suggestion
    async (err) => {
    // If it is here then err exists, right?
       return {success: false, error: err}
    }
  );

  return { success: true, error };
```

Reply:
done